src/postprocessing/idlparsed.js

```diff
@@ -8,17 +8,170 @@
+        // that composed IDL types ("interface mixin", "callback interface")
+        // only have the basic type in definitions.
+        dfnType = idl.type.split(' ')[0];
+        dfnName = idl.name;
+      }
+
+      const dfnNames = [];
+      if (dfnType === 'enum-value') {
+        // Bikeshed keeps wrapping quotes in the dfn linking text, not ReSpec.
+        dfnNames.push(dfnName);
+        dfnNames.push(`"${dfnName}"`);
+      }
+      else if (dfnType === 'method') {
+        // Bikeshed adds "..." for variadic arguments, not ReSpec. Let's try
+        // both variants. For overloads, Bikeshed essentially expects arguments
+        // to have different names, while ReSpec adds "!overload-x" to
+        // overloaded methods. We'll test all possibilities in order. If the
+        // spec only has a dfn for the most basic method, it's possible that we
+        // end up linking to that dfn from the overloaded methods too, but that
+        // seems good enough in practice.
+        // Last, method definitions sometimes appear without arguments (notably
+        // in the HTML spec).
+        const argsVariadic = member.arguments.map(arg => (arg.variadic ? '...' : '') + arg.name);
+        const args = member.arguments.map(arg => arg.name);
+        dfnNames.push(`${dfnName}!overload-${dfnOverload}(${args.join(', ')})`);
+        dfnNames.push(`${dfnName}(${argsVariadic.join(', ')})`);
+        dfnNames.push(`${dfnName}(${args.join(', ')})`);
+        dfnNames.push(`${dfnName}()`);
+      }
+      else if (dfnType === 'constructor') {
+        // Same as for methods
+        const argsVariadic = member.arguments.map(arg => (arg.variadic ? '...' : '') + arg.name);
+        const args = member.arguments.map(arg => arg.name);
+        dfnNames.push(`constructor!overload-${dfnOverload}(${args.join(', ')})`);
+        dfnNames.push(`constructor(${argsVariadic.join(', ')})`);
+        dfnNames.push(`constructor(${args.join(', ')})`);
+        dfnNames.push(`constructor()`);
+      }
+      else {
+        dfnNames.push(dfnName);
+      }
+
+      // Look for definitions that look like good initial candidates
+      const candidateDfns = spec.dfns
+        .filter(dfn => dfn.type === dfnType && !dfn.informative &&
+          (dfnFor ? dfn.for.includes(dfnFor) : true));
+
+      // Look for names in turn in that list of candidates.
+      for (const name of dfnNames) {
+        const dfns = candidateDfns.filter(dfn => dfn.linkingText.includes(name));
+        if (dfns.length > 0) {
+          if (dfns.length > 1) {
+            const forLabel = dfnFor ? ` for \`${dfnFor}\`` : '';
+            console.warn(`[warn] More than one dfn for ${dfnType} \`${dfnName}\`${forLabel} in [${spec.shortname}](${spec.crawled}).`);
+            return null;
+          }
+          else {
+            return dfns[0].href;
+          }
+        }
+      }
+
+      // Report missing dfns except for specs that we know already lack them
```

Comment:
How useful is this going to be moving forward? (i.e. how are likely are we going to detect new specs failing that pattern?) this may be best left to a strudy job?